internal/loader/configuration_setting_loader.go

```diff
@@ -535,26 +538,48 @@ func getConfigurationSettings(
 	nullString := "\x00"
 
 	for _, filter := range filters {
-		if filter.LabelFilter == nil {
-			filter.LabelFilter = &nullString // NUL is escaped to \x00 in golang
-		}
-		selector := azappconfig.SettingSelector{
-			KeyFilter:   &filter.KeyFilter,
-			LabelFilter: filter.LabelFilter,
-			Fields:      azappconfig.AllSettingFields(),
-		}
-		pager := client.NewListSettingsPager(selector, nil)
+		if filter.KeyFilter != nil {
+			if filter.LabelFilter == nil {
+				filter.LabelFilter = &nullString // NUL is escaped to \x00 in golang
+			}
+			selector := azappconfig.SettingSelector{
+				KeyFilter:   filter.KeyFilter,
+				LabelFilter: filter.LabelFilter,
+				Fields:      azappconfig.AllSettingFields(),
+			}
+			pager := client.NewListSettingsPager(selector, nil)
 
-		for pager.More() {
-			page, err := pager.NextPage(ctx)
+			for pager.More() {
+				page, err := pager.NextPage(ctx)
+				if err != nil {
+					e <- err
+				} else if len(page.Settings) > 0 {
+					c <- page.Settings
+				}
+			}
+		} else {
+			snapshot, err := client.GetSnapshot(ctx, *filter.SnapshotName, nil)
```

Comment:
Can we add snapshot status check since it can be archived? And what's expected behavior if  the snapshot expire after some interval?

Reply:
Archived snapshots are still available to be used.